src/main/java/com/dabsquared/gitlabjenkins/publisher/GitLabCommitStatusPublisher.java

```diff
@@ -44,7 +45,10 @@ public boolean prebuild(AbstractBuild<?, ?> build, BuildListener listener) {
     @Override
     public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
         Result buildResult = build.getResult();
-        if (buildResult == Result.SUCCESS) {
+        GitLabPushTrigger trigger = GitLabPushTrigger.getFromJob(build.getParent());
```

Comment:
The GitLabStatusPublisher is independent from the GitLabPushTrigger. You can add it to a job that hasn't the GitLabPushTrigger enabled so this would lead to a NullPointerException. Additionally I'd expect this setting within the Publisher itself and not in the settings of the trigger.